cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/ProgramControllerServiceAdapter.java

```diff
@@ -60,7 +67,11 @@ protected void doResume() throws Exception {
   @Override
   protected void doStop() throws Exception {
     if (service.state() != Service.State.TERMINATED && service.state() != Service.State.FAILED) {
+      LOG.debug("stopping service for program {}.", getProgramId());
       service.stopAndWait();
+      LOG.debug("stopped service for program {}, waiting for it to finish running listener hooks.", getProgramId());
```

Comment:
is this the program controller service? Perhaps the log message should say that?